### PR TITLE
Add provider model visibility and catalog refresh controls

### PR DESCRIPTION
**Pull request by sambitcreate**
## Summary
- add provider-wide Hide all / Show all visibility controls with safe all-hidden behavior
- add an explicit dual-source Update model catalogs action and bounded display-only models.dev cache
- refresh the bundled models.dev snapshot after main updates using read-only verification and credential-isolated publishing
- extend desktop, iOS, and Android regression coverage

## Review
- completed independent edge-case and adversarial reviews
- remediated hidden new-chat selection, policy-locked controls, validator parity, hard deadlines, loop guards, and CI credential exposure

## Verification
- npm test
- npm run test:model-catalog
- npm run type-check
- npm run lint
- focused Android and iOS model-catalog tests
- React Doctor changed-file scan

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -114,6 +114,8 @@
     "release:check-consumers": "node scripts/check-release-consumers.mjs",
     "release:verify": "node scripts/verify-macos-package.mjs --release",
     "models:refresh": "node scripts/update-model-capabilities.mjs",
+    "test:provider-model-catalog": "tsx --test renderer/components/settings/provider-model-visibility.test.tsx renderer/lib/model-picker-data.test.ts renderer/lib/use-model-selection.test.ts renderer/shared/gemini-usage-scope.test.ts main/services/config-store-core.test.ts main/services/aiden-remote-models.test.ts main/services/bot-capability-inventory-ports.test.ts",
+    "test:model-catalog": "npm run test:provider-model-catalog && tsx --test main/services/models-dev-cache-core.test.ts renderer/components/settings/providers-catalog-update.test.tsx main/services/models.test.ts scripts/model-snapshot-core.test.mjs && node --test scripts/update-model-capabilities.test.mjs scripts/check-ci-policy.test.mjs",
```
[the remark this answers is not in the document]
Comment (sambitcreate) [reply package.json:118]: Fixed in 161bbad68. PR CI now runs npm run test:model-catalog explicitly before the full npm test stage, and the CI-policy regression asserts that gate remains registered.

File: renderer/shared/model-visibility.ts [hunk 2]
```diff
@@ -14,33 +23,93 @@ function validIdentity(value: unknown): value is string {
...
-    total += models.length;
+    if (!validIdentity(providerId)) continue;
+
+    let defaultVisibility: ProviderModelVisibilityRule["defaultVisibility"] = "shown";
+    let policyHidden = false;
+    let rawExceptions: unknown = [];
+    if (Array.isArray(rawRule)) {
+      const legacy = rawRule.filter(validIdentity);
+      const wildcard = legacy.includes("*");
+      policyHidden = wildcard && providerId === "google";
```
[the remark this answers is not in the document]
Comment (sambitcreate) [reply renderer/shared/model-visibility.ts:50]: Fixed in 161bbad68. Settings normalization and runtime projection now derive the Google policy flag exclusively from geminiUsageScope; the regression covers both divergent directions and verifies an unrelated write persists the repaired canonical state.

**Comment by sambitcreate** [comment]
Follow-up: repository inspection confirms main currently has no repository ruleset and the classic branch-protection endpoint reports Branch not protected, so the catalog workflow bot push is permitted. Both actionable PullFrog findings are fixed in 161bbad68.
